src/main/java/it/hospital/project/appusers/UserService.java

```diff
@@ -0,0 +1,34 @@
+package it.hospital.project.appusers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserService {
+
+    UserRepository userRepository;
+
+    @Autowired
+    UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
```

Comment:
Autowired for construction injection is not neccesary

Reply:
Rafal, I kindly disagree. Basically construction injection is one of the good ways when autowiring a bean as long as it's done properly. The best practice in autowiring through a constructor should be like this:

e.g:

```
private final UserRepository userRepository;

@Autowired
public UserService(UserRepository userRepository) {

}
```

Pros:
Instance variable is will not be re-initialized.
The instance variable or the spring bean will always be required.

Cons:
Possible cluttering of arguments if in case it will be too many. It would be just a cosmetic coding issue. No performance overhead.

Reply:
Gem, I think you did not understand my thoughts. I am not against constructor injection, I only meant, that in this case @Autowired annotation is unnecessary. The less annotations you use, the better, IMHO.